### PR TITLE
doc: add cody quality tools + autocomplete review tool docs

### PR DESCRIPTION
**Pull request by slimsag**
Adds docs overview for cody quality tools (LLM judge, etc.) explaining what they are and how we can use them, plus a dedicated page for how to run/use the autocomplete review tool today.

## Test plan

Docs only

### CODE REVIEW

File: doc/dev/quality/autocomplete-review-tool.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    "endpoint": "https://sourcegraph-test-oai.openai.azure.com/",
+    "provider": "azure-openai",
+    "model": "text-embedding-ada-002-erik",
+    "dimensions": 1536
+  },
+```
+
+**Make sure that the Cody web interface shows you have working Chat and Embeddings on at least one repository before you continue.**
+
+## Overriding VS Code configuration
```
Comment (slimsag) [inline doc/dev/quality/autocomplete-review-tool.md:36]: This part is a bit nasty right now, I'm looking to improve it / make these always env vars.

File: doc/dev/quality/autocomplete-review-tool.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+# Autocomplete review tool
```
Comment (valerybugakov) [inline doc/dev/quality/autocomplete-review-tool.md:1]: Should we replace [this section](https://github.com/sourcegraph/cody/blob/main/vscode/CONTRIBUTING.md#running-the-autocomplete-test-suite) with a link to this doc?

**Review by valerybugakov — APPROVED** [review]
Thanks for adding this! 

Cc @umpox, since you plan to work on the autocomplete eval tooling.

File: doc/dev/quality/autocomplete-review-tool.md (hunk 2 above)
Comment (slimsag) [reply doc/dev/quality/autocomplete-review-tool.md:1]: Good idea 👍

**Pull request merged**